baseapp/baseapp.go

```diff
@@ -677,6 +677,9 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 	// Result if any single message fails or does not have a registered Handler.
 	result, err = app.runMsgs(runMsgCtx, msgs, mode)
 	if err == nil && mode == runTxModeDeliver {
+		// When block gas exceeds, it'll panic and won't commit the cached store.
+		consumeBlockGas()
```

Comment:
do we need to call it here? It's already called in `defer` (line 614)

Reply:
> do we need to call it here? It's already called in `defer` (line 614)

That's the main purpose of this PR, call it here so if block gas limit exceeded, it'll panic and won't execute the `msCache.Write`, so the tx msg side effects are reverted.